src/python/pants/backend/experimental/java/register.py

```diff
@@ -13,7 +18,13 @@
 
 
 def target_types():
-    return [JavaLibrary, JunitTests, JvmDependencyLockfile]
+    return [
+        JunitTestTarget,
+        JunitTestsGeneratorTarget,
+        JavaSource,
```

Comment:
Should probably be `JavaSourceTarget` to make clear this is not the field.

Reply:
Makes sense. I thought it was odd when I was going through the Shell example, but maybe I just missed it 🤷